README.md

```diff
@@ -37,6 +38,37 @@ Brion provides the following major features:
   [vmmlib](http://vmml.github.io/vmmlib),
   [Lunchbox](http://eyescale.github.io/Lunchbox-1.8/index.html),
   [Boost](http://www.boost.org/doc/libs).
+* Higher level classes:
+  * Read only morphology (brion::v2::Morphology)
+
+# Library versions {#Versions}
+
+Brion is evolving into a version 2 of its API. This new version will provide
+higher level classes to facilitate queries and manipulation of the data model
+provided by the low level IO objects. Version 2 is experimental at this
+moment, so it is not the default API and the major number of the library is
+still 1.
+
+## Version 1 compatibility
+
+If you are using the Brion 1 you don't need to do anything to compile your
+code against the latest version of Brion. However, the last release of
+version 1 is not binary compatible with the version 2 library, so
+you will need to recompile is you want to take advantage of the latest
+features that are backwards compatible.
+
+## Using version 2
+
+In order to use Brion 2 there is nothing special to be done to compile Brion
+itself. In client code you have to define the macro \c BRION_USE_V2_API to
+enable this version. When this macro is set, all the new high level classes
+will live in the namespace \c brion and classes which clash with new ones are
+moved into the \c brion::io If the macro is not enabled, version 2 classes
+will still be available inside the namespace \c brion::v2.
+
+There is no need to recompile Brion to switch between versions. It is
+guaranteed that this macro only changes the namespace layout and that class
+interfaces remain equal.
```

Comment:
Of course it's an SDK-like API, that's completely unavoidable because that's what it's trying to replace.
Since we didn't discuss how to deal with this I decided to follow an incremental approach to it, and actually that's what I thought @eile and me have agreed. A new library it's a good idea and for the moment I'd keep it in this project. I just hope it doesn't block this PR because we can't agree on a name for it. The new library can host the future SpikeReport.
It's painful that these questions emerge and are decided in PRs, you can do better but we need to spend more time in planning or involve more people.